app/lib/lnurl/types.ts

```diff
@@ -0,0 +1,71 @@
+export type LNURLError = {
+  status: 'ERROR';
+  reason: string;
+};
+
+/**
+ * Response from the lnurlp endpoint
+ * @see https://github.com/lnurl/luds/blob/luds/06.md
+ */
+export type LNURLPayParams = {
+  /**
+   * The type of the LNURL
+   */
+  tag: 'payRequest';
+  /**
+   * The URL from LN SERVICE which will accept the pay request parameters
+   * @example https://domain.com/api/lnurlp/callback/${user.id}
+   */
+  callback: string;
+  /**
+   * Max millisatoshi amount LN SERVICE is willing to receive
+   */
+  maxSendable: number;
+  /**
+   * Min millisatoshi amount LN SERVICE is willing to receive,
+   * can not be less than 1 or more than `maxSendable`
+   */
+  minSendable: number;
+  /**
+   * Metadata json which must be presented as raw string here,
+   * @example `[['text/plain', 'Pay to ${address}']]`
+   */
+  metadata: string;
```

Comment:
how about [comment](https://github.com/lnurl/luds/blob/luds/12.md)? do we not want that or not important for first iteration?